### ISSUE TITLE
findPlayer called from plex.sessions() failing on missing port in /status/sessions

### ISSUE THREAD

**Issue by shenning00** [mjs7231/python-plexapi#114]
I have some code that access current active sessions.

`sessions = plex.sessions()`

The session field 'player' is always 'None'.

Digging into it a bit I found that there is a util method findPlayer invoked:

```
def findPlayer(server, data):
    """ Returns the :class:`~plexapi.client.PlexClient` object found in the specified data.

        Parameters:
            server (:class:`~plexapi.server.PlexServer`): PlexServer object this is from.
            data (ElementTree): XML data to find Player in.
    """
    elem = data.find('Player')
    if elem is not None:
        from plexapi.client import PlexClient
        baseurl = 'http://%s:%s' % (elem.attrib.get('address'), elem.attrib.get('port'))
        return PlexClient(baseurl, server=server, data=elem)
    return None
```

It turns out that /status/sessions - Player section of the MediaContainer/Video does not contain a 'port' number.

<Player address="192.168.1.167" device="Roku 4" machineIdentifier="..." model="4400X" platform="Roku" platformVersion="7.x" product="Plex for Roku" profile="Roku-7.x" state="paused" title="Roku 4 - ..." vendor="Roku" version="4.3.0.3785-a10af3f-Plex"/>

So, in findPlayer where it builds the baseurl the port is None. This cause the PlexClient() to fail.

Let me know if I'm missing something.

**Comment by Hellowlol** [comment]
I just pushed a branch to github where i have "fixed" this. You can find the pr here: https://github.com/mjs7231/python-plexapi/pull/115

**Comment by shenning00** [comment]
@Hellowlol - not sure that this solves my problem.

Looks like you are querying the server /clients using server.clients(). 

For my setup /clients doesn't return the list of players. I find /clients very unpredictable.

Here's what I get today for /clients - 

```
<MediaContainer size="1">
<Server name="Plex Web (Chrome)" host=<PMS_IP> address=<PMS_IP> port="32400" machineIdentifier=<ID1> version="2.13.0" protocol="plex" product="Plex Web" deviceClass="pc" protocolVersion="1" protocolCapabilities="timeline,playback,navigation,mirror,playqueues"/>
</MediaContainer>
```

If I look at /status/sessions I get - 

```
<MediaContainer>
  <Video>
  ...
<Player address=<ROKU_IP> device="Roku 4" machineIdentifier=<ID2> model="4400X" platform="Roku" platformVersion="7.x" product="Plex for Roku" profile="Roku-7.x" state="playing" title="Roku 4 - ..." vendor="Roku" version="4.3.0.3785-a10af3f-Plex"/>
<Session id=<SESSION_ID> bandwidth="6359" location="lan"/>
  ...
  </Video>
</MediaContainer>
```

So, the Player listed in /status/sessions does not exist in /clients.

It does exist in plex.tv/devices.

**Comment by Hellowlol** [comment]
/clients is the best way to port.. What client is missing and is it connected to have same LAN as pms?

**Comment by mjs7231** [comment]
findPlayer is about to go away tonight as well.

**Comment by mjs7231** [comment]
More about this; I think we should be building the PlexClient when parsing sessions, but not actually connecting to the PlexClient at this point. Perhaps reload() or connect() should do that (at which point we can do as Hellowlol mentioned and parse /clients at that point to build the full object. It's a little tricky, but should work.

**Comment by shenning00** [comment]
@mjs7231 - I agree. It seems like there should be a separate connect() method.

**Comment by mjs7231** [comment]
This should be fixed on the 3.0.0 branch with commit c0958a0e. Basically, stop auto-connecting to Players unless we pass both a baseurl and token to PlexClient. We'll never have the information needed to make a successful connection anyway.

**Comment by mjs7231** [comment]
Closing this one. @Hellowlol wants to add a bit more functionality around connecting to clients, but the regression and bugs are gone (per the above comment). Everything else we do is an enhancement.
